Microsoft.DotNet.Interactive/Events/InputRequested.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using Microsoft.DotNet.Interactive.Commands;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Interactive.Events
+{
+    public class InputRequested : KernelEventBase
+    {
+        [JsonConstructor]
+        internal InputRequested(string prompt, bool password)
+        {
+            Prompt = prompt;
+            Password = password;
+        }
+
+        public InputRequested(
+            string prompt,
```

Comment:
prompt should be optional with default value

Reply:
What default value shall we use?

Reply:
Personally, I tend to avoid default parameter values for public APIs. I prefer a callsite to be explicit about the parameter it's using :)
But if this is the code style commonly used in dotnet-interactive, I will add default parameter values to them.

Reply:
I added the static method `input` to enable requesting input in C# and F#, and default parameter values are used there:
```
        public static string input(
            string prompt = "",
            bool isPassword = false)
```
Given that when calling this public constructor, we will pass in the `prompt` and `isPassword` values anyways, would it make more sense to not have default values here?